lib/base_service.js

```diff
@@ -92,6 +118,12 @@ class BaseService {
             this._logger = new Logger(config.logging);
             this._setUpDNSCaching();
 
+            // Catch unhandled rejections & log them. This relies on bluebird.
+            process.on('unhandledRejection', this._unhandledRejectionHandler);
```

Comment:
Wouldn't it be better to declare these right from the get-go instead of waiting for everything else to be set-up? Doing so would improve start-up logging itself too.

Reply:
Si, Agreed.